### ISSUE TITLE
[Slider] Add a `label` property to provide context to the component (CAL-AMP16) 

### ISSUE THREAD

**Issue by dqateam** [Esri/calcite-components#13529]
**monday.com sync:** #10821959597

### Violation: 
 Provide a valid label for form fields 

### WCAG Reference: 
- [2.5.3 Label in Name](https://www.w3.org/WAI/WCAG22/Understanding/label-in-name) 

### Severity: 
 Critical 

### Recommended solution
- [ ] Add a `label` property to slider to ensure its context is provided to AT
- [ ] **PE Discussion needed** do we need to no longer offer or recommend using an accompanying `calcite-label` (wrapped) with the component?
    - Depending on conversations we _may_ need to consider some documentation guidance for next steps 

### Media Type: 
 ARIA 

--- 
### Areas for Remediation: 
- Title: CAL-AMP16 Provide a valid label for form fields 
- Module: M3: Form validation_Calcite Feedback Survey 
- Finding ID: [CAL-AMP16](https://esri.hub.essentia11y.com/short-link/e0hjcd43GFeBns2x) 

**Issue**
There is a "slider" control (present below the "How many components are you familiar with?") with a visual label that is not programmatically associated with the control. 

**User Impact**
Screen reader users will have difficulty determining the purpose of these controls. Speech input users will have difficulty navigating to them.

**Code Reference** 
 ```
<div aria-orientation="horizontal" role="slider" tabindex="0" aria-valuemax="100" aria-valuemin="0" aria-valuenow="10"class="thumb thumb--value thumb--active" data-value-prop="value" style="right: 90%;">(...)</div>

```

**Steps to Reproduce**
1. Locate the form field.
2. Inspect it with Chrome DevTools.
3. In the Accessibility tab, expand the Computed Properties section.
4. Review the value for "Name".
 
**Recommendation**
Provide a valid label for form fields. The best way to do this is to use a ```<label>``` element with a for attribute set to the ID of the field it labels. For custom controls, use aria-labelledby on the control set to the ID of the labelling text.

**Compliant Code Example**
```
<!--Example using label element--> 
<label for="amount ">Amount:</label> <input id="Amount" type="text" name="amount" size="30"> 
---
<!--Example label has no "for" attribute but does wrap the input element)--> 
<label>Amount: <input id="amount " type="text" name="amount " size="30"></label> 
---
<!--Example using aria-labelledby--> 
<label id="lblAmount ">Amount:</label> <input aria-labelledby="lblAmount" id="Amount" type="text" name="amount" size="30"> 
---
<!-- Example using aria-label attribute - visual label exists but is not associated--> 
<span>Amount:</span> 
<input aria-label="Amount:" type="text" id="amount" size="30"> 
---
<!--Example using title attribute - visual label exists but is not associated--> 
<span>Amount:</span> 
<input title="Amount:" type="text" id="amount" size="30">
```

 **Issue Image** 
 https://desktopqa/aic_lap/Calcite/Calcite2025/CAL-AMP16.png 
 

 --- 
### Additional Resources: 
- [LA Platform Report Link](https://esri.hub.essentia11y.com/short-link/e0hjcd43GFeBns2x) 
- [Esri Accessibility Resources](https://esriis.sharepoint.com/sites/Accessibility)
- Report Source: Product accessibility evaluation conducted on Calcite by Level Access and delivered August, 2025.

**Comment by fdeters** [comment]
Just flagging that this is causing [axe aria-input-field-name](https://dequeuniversity.com/rules/axe/4.11/aria-input-field-name?application=axeAPI) violations in Oregon Metro's automated accessibility testing. Looking forward to this fix <3 

```
Promise rejected during "query parameters update correctly": The page should have no accessibility violations. Violations:
[serious]: ARIA input fields must have an accessible name 
Violated 1 time. Offending nodes are: 
<div aria-orientation="horizontal" role="slider" tabindex="0" aria-valuemax="1" aria-valuemin="0" aria-valuenow="0.5" class=" thumb thumb--value " data-value-prop="value" style="right:50%;"><!--?lit$885675480$--><!----><div class="handle"></div><!----><!----><!----></div>
```

**Comment by geospatialem** [comment]
~The above can be achieved using `aria-label="A custom label"`, although not fully integrated in Calcite.~ Moving the above to the backlog, but leaving the priority as `p - high` to tackle as soon as the team can prioritize the effort.


```html
<calcite-slider aria-label="A custom label"></calcite-slider>
```

**Comment by geospatialem** [comment]
Updating the [above comment](https://github.com/Esri/calcite-design-system/issues/13529#issuecomment-4434708135) as this does not fulfill all checks when conducting additional tests with the JS' implementation. Removed the `has workaround` label, and have this issue set for inclusion in R3.

**Comment by geospatialem** [comment]
Slider now offers a `label` property that provides context to AT, verified with `5.2.0-next.36` using NVDA and JAWS on Windows 🎉:

```html
<calcite-slider value="30" label="Park boundary layer transparency"></calcite-slider>
```

WRT the PE discussion, work was completed earlier this summer to add additional context to `label` and `labelText` with [Accessibility: Forms and labels](https://developers.arcgis.com/calcite-design-system/foundations/accessibility/#forms-and-labels) and across component samples.

Created a follow-up doc issue to review current `slider` component samples before `5.2`.
